packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts

```diff
@@ -0,0 +1,174 @@
+  private _authStackOptions: AmplifyAuthStackOptions | undefined; // options to help generate  cfn template
+  private _command: CommandType;
+  private _synthesizer: AuthStackSythesizer;
+  private _deploymentOptions: DeploymentOptions | undefined;
+  private _overrideProps: OverrideOptions | undefined;
+  private _cfnModifiers: Function | undefined;
+  private _authInputState: AuthInputState;
+
+  constructor(options: AmplifyAuthTransformOptions, command: CommandType) {
+    this._resourceConfig = options.resourceConfig;
+    this._command = command;
+    this._synthesizer = new AuthStackSythesizer();
+    this.app = new cdk.App();
+    this._deploymentOptions = options.deploymentOptions;
+    this._overrideProps = options.overrideOptions;
+    this._cfnModifiers = options.cfnModifiers;
+    this._authInputState = AuthInputState.getInstance({
+      category: this._resourceConfig.categoryName,
+      resourceName: this._resourceConfig.resourceName,
+      fileName: this._resourceConfig.stackFileName,
+    });
+  }
+
+  public async transform(): Promise<Template> {
+    // parse Input data
+    this._authStackOptions = await this.getInput();
+
+    // generate cfn Constructs and AmplifyRootStackTemplate object to get overridden
+    await this.generateAmplifyAuthStackTemplate();
+
+    // apply override on Amplify Object having CDK Constructs for Root Stack
+    await this.applyOverride();
+
+    // generate CFN template
+    const template: Template = await this.synthesizeTemplates();
+
+    // check if the cfnModifiers are defined for the Auth Stack
+    if (this._cfnModifiers) {
+      this._cfnModifiers(template);
+    }
+
+    // save stack
+    if (this._command === CommandType.ADD || this._command === CommandType.UPDATE) {
+      await this.deployOverrideStacksToDisk({
+        templateStack: template,
+        rootFilePath: this._deploymentOptions!.rootFilePath,
+      });
+    }
+    return template;
+  }
+
+  private applyOverride = async () => {
+    if (this._command === CommandType.ADD || this._command === CommandType.UPDATE) {
+      const { overrideProps } = await import(this._overrideProps!.overrideFnPath);
+      if (typeof overrideProps === 'function' && overrideProps) {
+        // await buildOverrideDir(this._overrideProps!.overrideDir);
+        this._authTemplateObj = overrideProps(this._authTemplateObj as AmplifyAuthStackTemplate);
+      } else {
+        console.log('There is no override setup yet for Root Stack. To enable override : Run amplify override root');
+      }
+    }
+  };
+  /**
+   *
+   * @returns Object required to generate Stack using cdk
+   */
+  private getInput = async (): Promise<AmplifyAuthStackOptions> => {
+    return {
+      authStackFileName: this._resourceConfig.stackFileName,
+      authStackInputPayload: this._authInputState.getCliInputPayload(),
+      event: CommandType.ADD,
+    };
+  };
+
+  /**
+   * Generates Root stack Template
```

Comment:
this is auth stack and not the root stack correct?

Reply:
yes